From 8373222c22baa6804be56caf664f89fbcb844dc7 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 25 Sep 2020 07:44:51 +0300
Subject: [PATCH] [SW-21029] goya2: ppboot: use correct dumpsel bit for pcie skip

This patch uses the DumpSel bit 2 to skip pcie initialization.

Change-Id: Ia1c96b754f3220b4756c11345d1b1499bde49822
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2/asic.c b/goya2/asic.c
index ddf76f2..972527c 100644
--- a/goya2/asic.c
+++ b/goya2/asic.c
@@ -16,11 +16,8 @@
 
 u32 hl_skip_pcie_init(void)
 {
-	/* No bootstrap bit for goya 2 to skip PCIe initialization-
-	 * TBD: identify boot strap bits for both M2 and Addin card variants.
-	 * Until then, skip PCIe Initialization, to facilitate power on.
-	 */
-	return 1;
+	/* Use DumpSel[2] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
 }
 
 u64 hl_pll_timestamp_freq_get(void)
